www/js/onboarding/OnboardingStack.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useContext } from "react";
+import { StyleSheet } from "react-native";
+import { AppContext } from "../App";
+import WelcomePage from "./WelcomePage";
+import ConsentPage from "./ConsentPage";
+import SurveyPage from "./SurveyPage";
+import SaveQrPage from "./SaveQrPage";
+import SummaryPage from "./SummaryPage";
+
+// true if loading/undetermined
+// 'welcome' if no config present
+// 'consent' if config present, but not consented
+// 'survey' if consented but intro not done
+// null if intro done
+const OnboardingStack = () => {
+
+  const { pendingOnboardingState } = useContext(AppContext);
+
+  console.debug('pendingOnboardingState in OnboardingStack', pendingOnboardingState);
+
+  if (pendingOnboardingState.route == 'welcome') {
+    return <WelcomePage />;
+  } else if (pendingOnboardingState.route == 'summary') {
+    return <SummaryPage />;
+  } else if (pendingOnboardingState.route == 'consent') {
+    return <ConsentPage />;
+  } else if (pendingOnboardingState.route == 'save-qr') {
+    return <SaveQrPage />;
+  } else if (pendingOnboardingState.route == 'survey') {
+    return <SurveyPage />;
+  } else {
+    return 'TODO'
+  }
+}
```

Comment:
This seems like a change in behavior from the previous onboarding flow. Concretely, the previous flow only tracked whether the study was joined and the intro was done. If the study was joined, and the intro was not done, we would always go to the summary page. With this, I think we will go to the page that we left off.

This seems like better UX overall, so I have no complaints. But I do know, that during testing, I have frequently killed the app and restarted during the onboarding to go back to a previous screen, since the previous onboarding flow did not have a back button either.

Does the new onboarding flow support a back button? I don't seem to see it in the styles here.

Reply:
Ah wait, it looks like this is purely in-memory state, so this will retain the previous behavior.
Feel free to resolve if that is true.